components/auth/org.wso2.carbon.auth.scope.registration.rest.api/src/test/java/org/wso2/carbon/auth/scope/registration/rest/api/impl/ScopesApiServiceImplTest.java

```diff
@@ -0,0 +1,154 @@
+    }
+
+    @Test
+    public void testRegisterExistingScope() throws Exception {
+        ScopesApiServiceImpl scopesApiService = getNewScopesApiServiceImpl();
+
+        //register a new Scope
+        ScopeDTO scopeDTO = createDefaultScopeDTO();
+        Response scopeAddResponse = scopesApiService.registerScope(scopeDTO, getNewMockedRequest());
+        Assert.assertEquals(scopeAddResponse.getStatus(), Response.Status.CREATED.getStatusCode());
+        Assert.assertTrue(scopeDTO.equals(scopeAddResponse.getEntity()));
+
+        //register same scope again and validate
+        scopeAddResponse = scopesApiService.registerScope(scopeDTO, getNewMockedRequest());
+        Assert.assertEquals(scopeAddResponse.getStatus(), Response.Status.CONFLICT.getStatusCode());
+    }
+
+    @Test
+    public void testGetInvalidScope() throws Exception {
+        ScopesApiServiceImpl scopesApiService = getNewScopesApiServiceImpl();
+        Response scopeGetResponse = scopesApiService.getScope("invalid-name", getNewMockedRequest());
+        Assert.assertEquals(scopeGetResponse.getStatus(), Response.Status.NOT_FOUND.getStatusCode());
+    }
+
+    @Test
+    public void testUpdateInvalidScope() throws Exception {
+        ScopesApiServiceImpl scopesApiService = getNewScopesApiServiceImpl();
+        Response updateResponse = scopesApiService
+                .updateScope(createDefaultScopeDTO(), "invalid-name", getNewMockedRequest());
+        Assert.assertEquals(updateResponse.getStatus(), Response.Status.NOT_FOUND.getStatusCode());
+    }
+
+    @Test
+    public void testDeleteInvalidScope() throws Exception {
+        ScopesApiServiceImpl scopesApiService = getNewScopesApiServiceImpl();
+        Response scopeDeleteResponse = scopesApiService.deleteScope("invalid-scope", getNewMockedRequest());
+        Assert.assertEquals(scopeDeleteResponse.getStatus(), Response.Status.NOT_FOUND.getStatusCode());
+    }
+
+    @Test
+    public void testCheckInvalidScopeExists() throws Exception {
+        ScopesApiServiceImpl scopesApiService = getNewScopesApiServiceImpl();
+        //validate the new scope exists
+        Response scopeExistsResponse = scopesApiService.isScopeExists("invalid-scope", getNewMockedRequest());
+        Assert.assertEquals(scopeExistsResponse.getStatus(), Response.Status.NOT_FOUND.getStatusCode());
+    }
+
+    @Test
+    public void testGetScopes() throws Exception {
+        ScopesApiServiceImpl scopesApiService = getNewScopesApiServiceImpl();
+
+        //register a new Scope
+        ScopeDTO scopeDTO = createDefaultScopeDTO();
+        Response scopeAddResponse = scopesApiService.registerScope(scopeDTO, getNewMockedRequest());
+        Assert.assertEquals(scopeAddResponse.getStatus(), Response.Status.CREATED.getStatusCode());
+        Assert.assertTrue(scopeDTO.equals(scopeAddResponse.getEntity()));
+
+        //register another new Scope
+        ScopeDTO scopeDTO2 = createAdditionalScopeDTO();
```

Comment:
We can have a generic method to create scopeDTO by providing the scope name and roles. No need to have multiple methods.

Reply:
Fixed, Thanks for the pointer.